paasta_tools/vitesscluster_tools.py

```diff
@@ -766,6 +802,8 @@ def get_vitess_config(self) -> VitessDeploymentConfigDict:
             vtadmin=self.get_vtadmin(),
             keyspaces=self.get_keyspaces(),
             updateStrategy=self.get_update_strategy(),
+            prometheusPushGateway=self.get_prometheus_pushgateway_config(),
+            vitessMetricScrapper=self.get_load_vitess_metrics_config(),
```

Comment:
note: these sidecars should probably also have resource requests

Reply:
question: will vitess understand this config? from looking at the vitess crd docs, it looks like a couple of their CRDs expose a sidecarContainers field that takes an array of k8s V1Container objects

Reply:
Vitess already exposes the metrics via and endpoint so the sidecars are more for Prometheus/Grafana, and Vitess itself I don't think needs to know about the configs specifcally just need the sidecars running for push the metrics and filter them per tablet via the bash script.

Reply:
correct, i'm suggesting that we do the prometheus translation with a single sidecar that we configure prometheus to scrape rather than have two sidecars (one for translation and another for actually getting the metrics into prometheus)

Reply:
@alexanderjulianmartinez I think this might not work as is. The vttablet spec (and similarly for vtctld and other vitess components) has a field to include a `sidecarContainers` list (https://github.yelpcorp.com/misc/compute-infra-k8s/blob/master/clusters/eks-infrastage/90-vitess-operator/eks/40-crd-planetscale.com-vitesskeyspaces.yaml#L564) of type core v1 container (https://github.com/planetscale/vitess-operator/blob/f1fe438f1a753007f5cace547b198a6e3cfaf1e3/pkg/operator/vttablet/spec.go#L60)

Core V1 Container json spec fields: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/

We'd ideally want to include the image and container spec for prometheus_pushgateway in the sidecarContainers with an init command if needed